widget/fileicon.go

```diff
@@ -161,9 +166,11 @@ func (s *fileIconRenderer) Refresh() {
 	if s.file.URI != s.file.cachedURI {
 		s.file.propertyLock.RLock()
 		s.file.setURI(s.file.URI)
```

Comment:
I know this is outside of this PR but I wonder if setURI should be called with the write lock as it changing fields, also cachedURI is read without a lock and written with a read lock